kong/plugins/aws-lambda/handler.lua

```diff
@@ -23,28 +27,71 @@ function AWSLambdaHandler:new()
   AWSLambdaHandler.super.new(self, "aws-lambda")
 end
 
-local function retrieve_parameters()
+local function retrieve_parameters(content_type, conf)
   ngx_req_read_body()
   local body_parameters, err
-  local content_type = ngx_req_get_headers()[CONTENT_TYPE]
   if content_type and string_find(content_type:lower(), "multipart/form-data", nil, true) then
     body_parameters = Multipart(ngx_req_get_body_data(), content_type):get_all()
-  elseif content_type and string_find(content_type:lower(), "application/json", nil, true) then
+  elseif content_type and string_find(content_type:lower(), "application/json", nil, true)
+         and not conf.request_body then
     body_parameters, err = cjson.decode(ngx_req_get_body_data())
     if err then
       body_parameters = {}
     end
-  else
+  elseif content_type and string_find(content_type:lower(), "application/x-www-form-urlencoded", nil, true) then
     body_parameters = public_utils.get_post_args()
+  else
+    body_parameters = {}
   end
 
   return utils.table_merge(ngx_req_get_uri_args(), body_parameters)
 end
 
+local function retrieve_body(content_type)
+  ngx_req_read_body()
+  local err, body, request
+  local rawBody = ngx_req_get_body_data()
+  if rawBody == nil then return nil end
+
+  if content_type and string_find(content_type:lower(), "application/json", nil, true) then
+    body, err = cjson.decode(rawBody)
+    if err then
+      return nil
+    end
+  elseif content_type and ( string_find(content_type:lower(), "application/xml", nil, true)
+    or  string_find(content_type:lower(), "application/soap+xml", nil, true)
+    or  string_find(content_type:lower(), "text/", nil, true) )
+  then
+    body = rawBody
+  else
+    body =  ngx_encode_base64( rawBody )
```

Comment:
there are certainly many content types that are not binary, that would end up Base64 encoded here. perhaps we should invert this logic, and only encode certain known content types, and raw pass everything else?